### PR TITLE
Update DNS Overview

### PR DESCRIPTION
**Pull request by JeremyLikness**
Added description, samples, example code. Tagging @camsoper for merge after team reviews and gives sign-off.

### CODE REVIEW

**Comment by openpublishbuild** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[api/overview/azure/dns.md](https://github.com/JeremyLikness/azure-docs-sdk-dotnet/blob/patch-4/api/overview/azure/dns.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/api/overview/azure/dns?branch=pr-en-us-275) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C7%5C31%5C14458122-e42f-10e3-bd3d-ee1c7494cfbf%5CPullRequest%5C201707311307045759-275%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: api/overview/azure/dns.md [hunk 1]
```diff
@@ -30,10 +33,30 @@ Install-Package Microsoft.Azure.Management.Dns.Fluent
 dotnet add package Microsoft.Azure.Management.Dns.Fluent
```
Comment (anthonychu) [inline api/overview/azure/dns.md]: The example below doesn't compile with the `Microsoft.Azure.Management.Dns.Fluent` package. It needs `Microsoft.Azure.Management.Dns`, which is only in preview.

File: api/overview/azure/dns.md [hunk 2]
```diff
@@ -30,10 +33,30 @@ Install-Package Microsoft.Azure.Management.Dns.Fluent
...
+```csharp
+/*
+using Microsoft.Azure.Management.Dns;
+using Microsoft.Azure.Management.Dns.Models;
+*/
+Zone dnsZoneParams = new Zone("global");
+dnsZoneParams.Tags = new Dictionary<string, string>();
+dnsZoneParams.Tags.Add("dept", "finance");
+Zone dnsZone =
+    await dnsClient.Zones.CreateOrUpdateAsync(resourceGroupName, zoneName, dnsZoneParams, null, "*");
```
Comment (anthonychu) [inline api/overview/azure/dns.md]: Might be helpful to include the code to create the client as well, as its constructor is a bit non-trivial.

Comment (JeremyLikness) [inline api/overview/azure/dns.md]: @anthonychu I made changes per your comments, please review and if it looks good sign-off or share further feedback, thanks!

**Comment by openpublishbuild** [comment]
### :x: Validation status: errors


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[api/overview/azure/index.yml](https://github.com/JeremyLikness/azure-docs-sdk-dotnet/blob/patch-4/api/overview/azure/index.yml) | :x:Error | | [Details](#user-content-api%2Foverview%2Fazure%2Findex.yml)
[]() | :x:Error | | [Details](#user-content-)

<a name="api%2Foverview%2Fazure%2Findex.yml"></a>
### [api/overview/azure/index.yml](https://github.com/JeremyLikness/azure-docs-sdk-dotnet/blob/patch-4/api/overview/azure/index.yml)
 - **[Error]** Error transforming model &quot;W:\uoa0-o-fb7a5e42\raw-\wshj0oo3.i33\rawmodel\api\overview\azure\index.raw.json&quot; generated from &quot;api/overview/azure/index.yml&quot; using &quot;ManagedReference.html.primary.js&quot;. Error running Transform function inside template preprocessor: 

<a name=""></a>
### []()
 - **[Error]** Error transforming model &quot;W:\uoa0-o-fb7a5e42\raw-\wshj0oo3.i33\rawmodel\api\overview\azure\index.raw.json&quot; generated from &quot;api/overview/azure/index.yml&quot; using &quot;ManagedReference.html.primary.js&quot;. Error running Transform function inside template preprocessor: 

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C8%5C1%5C14458122-e42f-10e3-bd3d-ee1c7494cfbf%5CPullRequest%5C201708011320553288-275%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by anthonychu** [comment]
#sign-off

**Pull request merged**